### PR TITLE
Add execinquery linter

### PR DESCRIPTION
**Pull request by lufeee**
execinquery is a linter that detects query at database/sql package Query function.
in detail, please check following link.
https://github.com/lufeee/execinquery

### CODE REVIEW

File: pkg/lint/lintersdb/manager.go [hunk 1]
```diff
@@ -684,6 +684,11 @@ func (m Manager) GetAllSupportedLinterConfigs() []*linter.Config {
 			WithSince("v1.26.0").
 			WithPresets(linter.PresetStyle).
 			WithURL("https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/nolintlint/README.md"),
+
+		linter.NewConfig(golinters.NewExecinQuery()).
+			WithSince("1.0.0").
```
Comment (sivchari) [inline pkg/lint/lintersdb/manager.go:689]:
```suggestion
			WithSince("1.46.0").
```

File: test/testdata/querycheck.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	"database/sql"
+)
+
+func main() {
+	db, _ := sql.Open("mysql", "test:test@tcp(test:3306)/test")
+
+	test := "a"
+	_, err = db.Query("Update * FROM hoge where id = ?", test) // ERROR "Query not `SELECT` query"
```
Comment (sivchari) [inline test/testdata/querycheck.go:11]: Please add more test case. (e.g. QueryContext, Execute, ExecuteContext).

**Comment by sivchari** [comment]
Thanks for contributing. I commented nits.

**Comment by ldez** [comment]
Could you rebase your PR? Thanks.

**Comment by lufeee** [comment]
Sorry, I don't get the meaning.
rebase means squash the all commit, isn't it?

**Comment by ldez** [comment]
> Sorry, I don't get the meaning.

can you rebase your branch (related to your Pull Request) on the branch `master` of golangci-lint?

> rebase means squash the all commit, isn't it?

nope, rebase means just [rebase](https://git-scm.com/docs/git-rebase)

example:

```
git rebase upstream/master
```

**Comment by lufeee** [comment]
Thanks, i got it.

**Comment by ldez** [comment]
I will rebase your branch because I feel that you need help.

**Comment by ldez** [comment]
Your tag (v1.0) is not valid from the point of view of Go, can you create a valid tag (v1.0.0)?

**Comment by lufeee** [comment]
Thanks a lot!

**Comment by lufeee** [comment]
I added new 'v1.0.0' tag and Fix analyzer name

**Comment by sivchari** [comment]
Can you update go modules ? Because this PR has used incorrect tag version's linter, yet.

File: go.mod [hunk 3]
```diff
@@ -51,6 +52,7 @@ require (
 	github.com/ldez/gomoddirectives v0.2.2
 	github.com/ldez/tagliatelle v0.3.1
 	github.com/leonklingele/grouper v1.1.0
```
Comment (sivchari) [inline go.mod:54]: Please replace v0.0.0-20220322150457-8f5c1333d66e to v1.0.0.

**Comment by ldez** [comment]
For your information:
- to update the go.sum: `go mod tidy`
- a direct dependency must not be flagged as indirect.

**Comment by lufeee** [comment]
> For your information:
> 
> * to update the go.sum: `go mod tidy`
> * a direct dependency must not be flagged as indirect.

Thanks for sharing your best solution about dependence.

**Review by sivchari — APPROVED** [review]
LGTM

File: pkg/golinters/execinquery.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	"golang.org/x/tools/go/analysis"
+
+	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
+)
+
+func NewExecInQuery() *goanalysis.Linter {
+	const linterName = "execinquery"
+
+	a := execinquery.Analyzer
+	a.Name = linterName // TODO the name must change inside the linter.
```
Comment (sivchari) [inline pkg/golinters/execinquery.go:14]:
```suggestion
	a.Name = linterName
```

**Review by sivchari — CHANGES_REQUESTED** [review]
Sorry, one nits.

**Comment by sivchari** [comment]
Thanks @ldez .
LGTM

**Review by sivchari — APPROVED** [review]

**Pull request merged**